tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/DefaultModelBindingContext_SetResult_Integration.cs

```diff
@@ -39,6 +41,27 @@ namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore
     IntegrationName = IntegrationName,
     CallTargetIntegrationType = IntegrationType.Derived,
     InstrumentationCategory = InstrumentationCategory.AppSec)]
+    [InstrumentMethod(
+    AssemblyName = "Microsoft.AspNetCore.Mvc.Core",
+    TypeName = "Microsoft.AspNetCore.Mvc.ModelBinding.DefaultModelBindingContext",
+    MethodName = "set_Result",
+    ReturnTypeName = ClrNames.Void,
+    ParameterTypeNames = new[] { "Microsoft.AspNetCore.Mvc.ModelBinding.ModelBindingResult" },
+    MinimumVersion = "2.0.0.0",
+    MaximumVersion = "7.*.*.*.*",
+    IntegrationName = IntegrationName,
+    InstrumentationCategory = InstrumentationCategory.Iast)]
```

Comment:
Hmm, yeah, this seems a bit weird tbh, we're sending the exact same definition to instrument to the backend _4 times_. That seems both wasteful and confusing 🤔 

- In `Security you're calling the same `NativeMethods.InitializeProfiler` twice, once with the non-derived instrumentations, and once with the derived instrumentations. Why? (for the tracer, we call a _different_ initialization method, `NativeMethods.AddDerivedInstrumentations` with the derived ones. This looks like it could be a bug in AppSec to me?
- I haven't checked, but what's the result of sending the same definition multiple times (once for appsec, once for iast)? Do we have any tests of this?
-

Reply:
I checked that it was running fine when doing manual testing, but I have added a new integration test that actually taints the body with both Iast and AppSec enabled to make sure.

Reply:
>In Security you're calling the same NativeMethods.InitializeProfilertwice, once with the non-derived instrumentations, and once with the derived instrumentations. Why? (for the tracer, we call a _different_ initialization method,NativeMethods.AddDerivedInstrumentations` with the derived ones. This looks like it could be a bug in AppSec to me?

Thanks for the note, I think at the time, I wasn't aware of this method, and we should change it similarly to the tracer.. Ooc, is there a difference between doing the two? I'll investigate further also and open a PR related

Reply:
> is there a difference between doing the two

Yes, I believe so - as you're not calling  `AddDerivedInstrumentations` you're not _actually_ instrumenting derived implementations! 😬 Which also suggests there's no integration test using a derived version, so might be worth adding one? (Which will be a real PITA so maybe not 😉)